src/storage-client/src/types/connections.rs

```diff
@@ -867,6 +877,54 @@ impl RustType<ProtoTunnel> for Tunnel {
     }
 }
 
+impl Tunnel {
+    /// Setup and ssh tunnel to `remote_host:remote_port`, returning the local port
+    /// and a drop token for the session. This method is on `Tunnel` for convenience,
+    /// but panics if the `Tunnel` is not `Tunnel::Ssh`
```

Comment:
Let's split out a separate SSH tunnel struct, as in

```rust
enum Tunnel {
    Ssh(SshTunnel)
}

struct SshTunnel {
    connection: ...,
    connection_id: GlobalId,
}
```

so that there's a natural place to stick this method, and we don't have to explain the situation here!